### PR TITLE
Jon/linting

### PR DESCRIPTION
**Pull request by jonborchardt**
this is the config needed (and some example conversions)
i have punted on making this run as part of the build.
we can add that later

if we like this, ill fix up the project add add fixed files

soon, we will add a ci task to validate this is run before checkin

### CODE REVIEW

File: @types/images.d.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
-declare module "*.svg" {
+declare module '*.svg' {
```
Comment (jonborchardt) [inline @types/images.d.ts]: ignore for review, example of auto fix

File: components/link/base.ts [hunk 2]
```diff
@@ -6,21 +6,21 @@ export interface BaseLinkProps {
...
 export function baseLinkStyles(_: BaseLinkProps) {
-    return (
-        css<BaseLinkProps>`
-            color: ${({ theme, active, contrast }) => (
-                active
-                ? (contrast ? theme.link.contrastActiveColor : theme.link.activeColor)
-                : (contrast ? theme.link.contrastColor : theme.link.color)
-            )};
-            text-decoration: ${({theme}) => theme.link.decoration};
+    return css<BaseLinkProps>`
```
Comment (jonborchardt) [inline components/link/base.ts]: ignore for review, example of auto fix

File: demo/src/AppRoute.ts [hunk 3]
```diff
@@ -20,7 +20,7 @@ export interface AppRoute {
     disabled?: boolean;
 
     /* an optional tag we can display next to the link, eg. NEW */
-    tag?: {label: string, color: Color };
+    tag?: { label: string; color: Color };
```
Comment (jonborchardt) [inline demo/src/AppRoute.ts]: ignore for review, example of auto fix

File: postcss.config.js [hunk 4]
```diff
@@ -3,7 +3,5 @@
  * for distribution.
  */
 module.exports = {
-    plugins: [
-        require('cssnano')({ preset: 'default' })
-    ]
+    plugins: [require('cssnano')({ preset: 'default' })]
```
Comment (jonborchardt) [inline postcss.config.js]: ignore for review, example of auto fix

File: theme/breakpoints.ts [hunk 5]
```diff
@@ -8,4 +7,4 @@ export const breakpoints = {
     lg: Spacing.fromPixels('lg', 992),
     xl: Spacing.fromPixels('xl', 1200),
     xxl: Spacing.fromPixels('xxl', 1600)
-}
+};
```
Comment (jonborchardt) [inline theme/breakpoints.ts]: ignore for review, example of auto fix

File: theme/button.ts [hunk 6]
```diff
@@ -29,8 +29,10 @@ export const defaultSpacing = [spacing.md, spacing.xs];
 const marketingSpacing = [spacing.lg, spacing.sm];
 const paddingWithBorderInside = (borderWidth: number, initialPadding = defaultSpacing) => {
     // initial pad minus the border width + a single pixel for the standard border width expected by other components
-    return `${convertPixelsToRem(initialPadding[1].getPxValue() - borderWidth + 1)} ${convertPixelsToRem(initialPadding[0].getPxValue() - borderWidth + 1)}`;
-}
+    return `${convertPixelsToRem(
```
Comment (jonborchardt) [inline theme/button.ts]: ignore for review, example of auto fix

File: theme/colors.ts [hunk 7]
```diff
@@ -11,6 +11,7 @@ export class Color {
         this.hex = hex.toUpperCase();
         this.rgb = hexToRgb(hex);
     }
+
```
Comment (jonborchardt) [inline theme/colors.ts]: ignore for review, example of auto fix

File: theme/spacing.ts [hunk 8]
```diff
@@ -5,15 +5,19 @@ export class Spacing {
     static fromPixels(displayName: string, px: number) {
         return new Spacing(displayName, `${px}px`, convertPixelsToRem(px));
     }
+
```
Comment (jonborchardt) [inline theme/spacing.ts]: ignore for review, example of auto fix

File: theme/theme.less.ts [hunk 9]
```diff
@@ -132,8 +132,12 @@ export const less = `
     @primary-2: ${color.B2};
     @primary-3: ${color.B3};
     @primary-4: ${color.B4};
-    @primary-5: ${DefaultVarnishTheme.palette.primary.light}; // color used to control the text color in many active and hover states
-    @primary-6: ${DefaultVarnishTheme.palette.primary.main}; // color used to control the text color of active buttons
+    @primary-5: ${
```
Comment (jonborchardt) [inline theme/theme.less.ts]: ignore for review, example of auto fix

File: theme/typography.ts [hunk 10]
```diff
@@ -10,46 +10,64 @@ const fontWeightLight = fontWeight.light;
 const fontWeightRegular = fontWeight.regular;
 const fontWeightMedium = fontWeight.medium;
 const fontWeightBold = fontWeight.bold;
-const fallbackDefaultFonts = ['Helvetica Neue', 'Helvetica', '-apple-system', 'BlinkMacSystemFont',
-    'Segoe UI', 'PingFang SC', 'Hiragino Sans GB', 'Microsoft YaHei', 'Arial', 'sans-serif',
-    'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol'];
-const fallbackDefaultCodeFonts = ['SFMono-Regular', 'Consolas', 'Liberation Mono', 'Menlo',
-    'Courier', 'monospace'];
+const fallbackDefaultFonts = [
```
Comment (jonborchardt) [inline theme/typography.ts]: ignore for review, example of auto fix

File: utils/base.ts [hunk 11]
```diff
@@ -1,7 +1,12 @@
-export type Dictionary<T> = { [id:string]: T };
+export type Dictionary<T> = { [id: string]: T };
```
Comment (jonborchardt) [inline utils/base.ts]: ignore for review, example of auto fix

File: package.json [hunk 12]
```diff
@@ -6,6 +6,8 @@
   "license": "Apache-2.0",
   "repository": "git@github.com:allenai/varnish.git",
   "scripts": {
+    "lint": "eslint '**/*.{js,ts,tsx,json,yml,yaml}' && echo 'Lint complete.'",
```
Comment (codeviking) [inline package.json]: ESLint can lint YAML files?

Also, given the `ts` files are compiled to `js`, maybe we should ignore those?

Comment (jonborchardt) [reply package.json]: we do ignore the compiled js files, but this catches any js we add manually....

**Pull request merged**